examples/systemcalls/systemcalls.c

```diff
@@ -84,14 +138,65 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     // and may be removed
     command[count] = command[count];
 
+    int fd = open(outputfile, O_WRONLY | O_TRUNC | O_CREAT, 
+                S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+                    
+    if (fd == -1) 
+    {
+        /*error opening file*/
+        perror("open");
+        return false;
+    }
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    pid = fork();
```

Comment:
@sota6640 think about how you could share logic between this function and `do_exec`.